src/ProjectSystemTools/BuildLogging/Model/Frontend/UIBuildSummary.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.ProjectSystem.Tools.Providers;
+using Microsoft.VisualStudio.ProjectSystem.Tools.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model.Frontend
+{
+    /// <summary>
+    /// Immutable Type
+    /// </summary>
+    public sealed class UIBuildSummary : IBuildSummary
+    {
+        public int BuildID { get; }
+        public BuildType BuildType { get; }
+
+        public IEnumerable<string> Dimensions { get; }
+
+        public IEnumerable<string> Targets { get; }
+
+        public DateTime StartTime { get; }
+
+        public TimeSpan Elapsed { get; }
+
+        public BuildStatus Status { get; }
+
+        public string ProjectPath { get; }
+
+        public UIBuildSummary(int buildID, string projectPath, IEnumerable<string> dimensions, IEnumerable<string> targets, BuildType buildType, DateTime startTime)
+        {
+            BuildID = buildID;
+            ProjectPath = projectPath;
+            Dimensions = dimensions.ToArray();
+            Targets = targets?.ToArray() ?? Enumerable.Empty<string>();
+            BuildType = buildType;
+            StartTime = startTime;
+            Status = BuildStatus.Running;
+        }
+        public UIBuildSummary(IBuildSummary other, BuildStatus status, TimeSpan elapsed)
+        {
+            BuildID = other.BuildID;
+            BuildType = other.BuildType;
+            // TODO: Check if this needs deep copying
+            Dimensions = other.Dimensions;
+            Targets = other.Targets;
+            StartTime = other.StartTime;
+            ProjectPath = other.ProjectPath;
+
+            Elapsed = elapsed;
+            Status = status;
+        }
```

Comment:
These constructors are unused and can be removed, unless you're planning on using them in future. Even still, I would add them at such a time, in case your approach changes and these don't get cleaned up.